File: backend/app/models/chapter.py
from sqlalchemy import Column, Integer, String, Text, ForeignKey
from sqlalchemy.orm import relationship
from app.db.base import Base

class Chapter(Base):
    __tablename__ = "chapters"

    id = Column(Integer, primary_key=True, index=True)
    key = Column(String(50), unique=True, nullable=False)
    title = Column(String(200), nullable=False)
    sort_order = Column(Integer, default=0)

    knowledge_points = relationship("KnowledgePoint", back_populates="chapter", cascade="all, delete-orphan")

class KnowledgePoint(Base):
    __tablename__ = "knowledge_points"

    id = Column(Integer, primary_key=True, index=True)
    key = Column(String(50), unique=True, nullable=False)
    title = Column(String(200), nullable=False)
    content = Column(Text, default="")
    guide = Column(Text, default="")
    default_code = Column(Text, default="")
    sort_order = Column(Integer, default=0)
    chapter_id = Column(Integer, ForeignKey("chapters.id"), nullable=False)

    chapter = relationship("Chapter", back_populates="knowledge_points")

File: frontend/src/api/chapter.ts
import request from './request'

export interface KnowledgePoint {
  id: number
  key: string
  title: string
  content: string
  guide: string
  default_code: string
  sort_order: number
  chapter_id: number
}

export interface Chapter {
  id: number
  key: string
  title: string
  sort_order: number
  knowledge_points: KnowledgePoint[]
}

export const chapterApi = {
  list: () => request.get<Chapter[]>('/v1/chapters/'),
  create: (data: { key: string; title: string; sort_order?: number }) =>
    request.post<Chapter>('/v1/chapters/', data),
  update: (id: number, data: Partial<{ title: string; sort_order: number }>) =>
    request.put<Chapter>(`/v1/chapters/${id}`, data),
  delete: (id: number) => request.delete(`/v1/chapters/${id}`),

  createPoint: (chapterId: number, data: { key: string; title: string; content?: string; sort_order?: number }) =>
    request.post<KnowledgePoint>(`/v1/chapters/${chapterId}/points`, data),
  updatePoint: (pointId: number, data: Partial<{ title: string; content: string; guide: string; default_code: string; sort_order: number }>) =>
    request.put<KnowledgePoint>(`/v1/chapters/points/${pointId}`, data),
  deletePoint: (pointId: number) => request.delete(`/v1/chapters/points/${pointId}`),
}
